vscode/src/qirGeneration.ts

```diff
@@ -30,26 +30,22 @@ export async function getQirForActiveWindow(): Promise<string> {
     );
   }
 
-  // Check that the current target is base profile, and current doc has no errors.
+  // Check that the current target is base or adaptive_ri profile, and current doc has no errors.
   const targetProfile = getTarget();
-  const enablePreviewQirGen = getEnablePreviewQirGen();
-  if (targetProfile !== "base") {
-    const allowed = targetProfile === "adaptive_ri" && enablePreviewQirGen;
-    if (!allowed) {
-      const result = await vscode.window.showWarningMessage(
-        "Submitting to Azure is only supported when targeting the QIR base profile.",
-        { modal: true },
-        { title: "Change the QIR target profile and continue", action: "set" },
-        { title: "Cancel", action: "cancel", isCloseAffordance: true },
+  if (targetProfile === "unrestricted") {
+    const result = await vscode.window.showWarningMessage(
+      "Submitting to Azure is only supported when targeting the QIR base or adaptive_ri profile.",
+      { modal: true },
+      { title: "Change the QIR target profile and continue", action: "set" },
```

Comment:
I'd have to defer to @minestarks on this.

Reply:
Yeah, this is doable but probably out of scope. Maybe for now the button can say "Set the QIR target profile to _Base_ and continue" or something, so that it's clear what it's doing.